### PR TITLE
jgrss/safe tarfile extraction

### PR DESCRIPTION
**Pull request by jgrss**
## What is this PR changing?
Adds safe tarfile extraction following #229 
  
### Checklist
- [ ] Remember to add a semantic tag to the commit name
  
Tag options:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
  - refactor: A code change that neither fixes a bug nor adds a feature
  - perf: A code change that improves performance
  - test: Adding missing or correcting existing tests
  - chore: Changes to the build process or auxiliary tools and libraries such as documentation generation

Example:
```
fix: <branch name> PR number
```

### CODE REVIEW

File: src/geowombat/util/web.py [hunk 1]
```diff
@@ -726,14 +793,35 @@ def get_sentinel2_urls(safe_id, bands=None, cloud='gcp'):
...
+        scene_urls = [
+            f'{gcp_base}/tiles/{lid}_B{band_pos:02d}.jp2' for band_pos in band_pos
+        ]
         meta_url = f'{utm}/{zone}/{id_}/{safe_id}/MTD_TL.xml'
 
         return scene_urls, meta_url
 
 
-class GeoDownloads(CloudPathMixin, DownloadMixin):
+def _is_within_directory(directory: Path, target: Path) -> bool:
```
Comment (jgrss) [inline src/geowombat/util/web.py:804]: New function

File: src/geowombat/util/web.py [hunk 2]
```diff
@@ -726,14 +793,35 @@ def get_sentinel2_urls(safe_id, bands=None, cloud='gcp'):
...
+    abs_directory = str(directory.absolute())
+    abs_target = str(target.absolute())
+
+    prefix = os.path.commonprefix([abs_directory, abs_target])
+
+    return prefix == abs_directory
+
+
+def _safe_extract(
```
Comment (jgrss) [inline src/geowombat/util/web.py:813]: New function

**Comment by jgrss** [comment]
@mmann1123 I commented on the two new functions. The remainder is Black formatting.

**Review by mmann1123 — APPROVED** [review]
Glad we are resolving this. All yours

**Pull request merged**